packages/@aws-cdk/aws-codepipeline/lib/jenkins-provider.ts

```diff
@@ -0,0 +1,256 @@
+import cpapi = require('@aws-cdk/aws-codepipeline-api');
+import cdk = require('@aws-cdk/cdk');
+import { CustomActionRegistration } from "./custom-action-registration";
+import {
+  CommonJenkinsBuildActionProps,
+  CommonJenkinsTestActionProps,
+  JenkinsBuildAction,
+  JenkinsTestAction
+} from "./jenkins-actions";
+
+/**
+ * A Jenkins provider.
+ *
+ * If you want to create a new Jenkins provider managed alongside your CDK code,
+ * instantiate the {@link JenkinsProvider} class directly.
+ *
+ * If you want to reference an already registered provider,
+ * use the {@link JenkinsProvider#import} method.
+ */
+export interface IJenkinsProvider {
+  readonly providerName: string;
+  readonly serverUrl: string;
+  readonly version: string;
+
+  _includeBuildVariant(): void;
+  _includeTestVariant(): void;
+
+  /**
+   * Convenience method for creating a new {@link JenkinsBuildAction},
+   * and adding it to the given Stage.
+   *
+   * @param stage the Pipeline Stage to add the new Action to
+   * @param name the name of the newly created Action
+   * @param props construction properties of the new Action
+   * @returns the newly created {@link JenkinsBuildAction}
+   */
+  addToPipeline(stage: cpapi.IStage, name: string, props: CommonJenkinsBuildActionProps):
+      JenkinsBuildAction;
+
+  /**
+   * Convenience method for creating a new {@link JenkinsTestAction},
+   * and adding it to the given Stage.
+   *
+   * @param stage the Pipeline Stage to add the new Action to
+   * @param name the name of the newly created Action
+   * @param props construction properties of the new Action
+   * @returns the newly created {@link JenkinsTestAction}
+   */
+  addToPipelineAsTest(stage: cpapi.IStage, name: string, props: CommonJenkinsTestActionProps):
+      JenkinsTestAction;
+}
+
+/**
+ * Properties for importing an existing Jenkins provider.
+ */
+export interface ImportJenkinsProviderProps {
```

Comment:
We're renaming these to `JenkinsProviderImportProps`.

Reply:
Done.